test/comps/test_util.cc

```diff
@@ -26,68 +26,126 @@ class TestUtil : public ::testing::Test
 
     virtual void TearDown() {};
 
+    template <typename T>
+    struct UtilTestData {
+        std::vector<T> A;
+        std::vector<T> A_cpy;
+        std::vector<T> s;
+        std::vector<T> A_norm;
+
+        UtilTestData(int64_t m, int64_t n) :
+        A(m * n, 0.0), 
+        A_cpy(m * n, 0.0), 
+        s(n, 0.0),
+        A_norm(m * n, 0.0) 
+        {}
+    };
+
```

Comment:
This is kind of weird. I think it's unlikely that all tests of utilities will need ``A_cpy`` or ``A_norm``, yet you're allocating both of them for every test. I see how you're mimicking the structure of ``QBTestData``, but that isn't really appropriate here. Think about the differences in context: QB exists in the context of RandLAPACK's highly configurable algorithms-as-objects API, while utility functions are mundane and handle very different tasks.

Reply:
Fixed.